packages/react/src/menu/popup/MenuPopup.tsx

```diff
@@ -75,7 +76,10 @@ const MenuPopup = React.forwardRef(function MenuPopup(
       context={floatingContext}
       modal={false}
       initialFocus={nested ? -1 : 0}
+      returnFocus
       disabled={!mounted}
+      visuallyHiddenDismiss={modal ? 'Dismiss popup' : undefined}
+      order={!nested ? ['content', 'reference'] : undefined}
```

Comment:
What's `order` for? I notice the popup doesn't take focus on click with a pointer - not sure if this matters though.

Reply:
This makes the trigger not receive the data-floating-ui-inert attribute, so it can properly fire its click handler. Without it, it becomes inert (via poiner-events) after mousedown, so mouseup doesn't fire.

Reply:
I think it's a good idea to force `initialFocus` on the Popup if it's `undefined` now. It removes the default `:focus-visible` outline on the trigger

Reply:
Also, shouldn't select's trigger need it?

Reply:
> Also, shouldn't select's trigger need it?

It doesn't have the click handler, so we don't need it, but it might be a good idea anyway to make external click events work

Reply:
> I think it's a good idea to force initialFocus on the Popup if it's undefined now. It removes the default :focus-visible outline on the trigger

Do you mean `initialFocus={nested ? -1 : popupRef}`?

Reply:
Something like that yes